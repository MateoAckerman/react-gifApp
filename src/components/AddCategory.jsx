import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {
    // export const AddCategory = ({ onAddCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const inputValueTrim = inputValue.trim();

        if (inputValueTrim.length <= 1) return;
        onNewCategory(inputValueTrim);
        setInputValue('');

    }

    return (
        <form onSubmit={(event) => onSubmit(event)}>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            // onChange={e => setInputValue(e.target.value)}
            />
        </form>

    )
}
