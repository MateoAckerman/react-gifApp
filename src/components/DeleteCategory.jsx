import { useState } from "react"

export const DeleteCategory = ({ category, onDeleteCategory }) => {

    console.info("props button", category)

    // const onClick = (event) => {
    //     event.preventDefault();
    // }

    return (
        <button onClick={() => onDeleteCategory(category)}> Eliminar Categoria </button>
    )
}