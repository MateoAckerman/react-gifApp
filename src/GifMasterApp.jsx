import { useState } from "react"
import { AddCategory, GifGrid } from "./components";


export const GifMasterApp = () => {

    const [categories, setCategories] = useState(['Sword Art Online']);
    const [newCategory, setNewCategory] = useState('');
    const [deletedCategory, setDeletedCategory] = useState('');

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) {
            return;
        }
        setCategories(prevCategories => [newCategory, ...prevCategories]);
        setNewCategory('');

    };

    const onDeleteCategory = (deletedCategory) => {
        console.info("categoria fuera", deletedCategory)
        const filteredCategories = categories.filter((item) => item !== deletedCategory)
        setCategories(filteredCategories)
    };

    console.log(categories);

    return (
        <>
            <h1>Gif Master App</h1>

            <AddCategory
                onNewCategory={onAddCategory}  // SAME >>> event => onAddCategory(event)
            />

            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category}
                        categories={categories}
                        onDeleteCategory={onDeleteCategory} />
                )
                )
            }

        </>
    )
}
