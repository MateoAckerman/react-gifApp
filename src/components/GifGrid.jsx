import { useEffect, useState } from 'react';

import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { DeleteCategory } from './DeleteCategory';
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category, categories, onDeleteCategory }) => {

    const { images, isLoading } = useFetchGifs(category);

    // console.log({ categories })


    return (
        <>
            <div className='gifGridHeaderItem'>
                <h3>{category}</h3>

                <DeleteCategory
                    key={category}
                    category={category}
                    onDeleteCategory={onDeleteCategory}
                />
            </div>


            {/* <LoadingMessage isLoading={isLoading} /> */}
            {
                isLoading && (<h2>Cargando Gifs ...</h2>) // = If isLoading = true -> renderiza el <h2></h2>

            }

            <div className='card-grid'>
                {
                    images.map((image) => (
                        <GifItem
                            key={image.id}
                            // De esta forma pasamos al componente GifItem las properties de image con el operador spread.
                            {...image}
                        />
                    ))
                }
            </div>



        </>
    )
}
