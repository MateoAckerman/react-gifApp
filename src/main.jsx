import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifMasterApp } from './GifMasterApp.jsx';

import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifMasterApp />
  </React.StrictMode>
);
