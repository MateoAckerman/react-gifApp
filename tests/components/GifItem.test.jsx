import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Pruebas en <GifItem />', () => {
    const titleInit = 'Sword Art Online';
    const urlInit = 'https://sao.dev/';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={titleInit} url={urlInit} />)
        expect(container).toMatchSnapshot();
    })
    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={titleInit} url={urlInit} />);
        screen.debug();
        // expect(screen.getByRole('img').src).toBe(urlInit);
        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(urlInit);
        expect(alt).toBe(titleInit);
    })
    test('Debe mostrar el parrafo con el titulo indicado', () => {
        render(<GifItem title={titleInit} url={urlInit} />);
        expect(screen.getByText(titleInit))
    })
})