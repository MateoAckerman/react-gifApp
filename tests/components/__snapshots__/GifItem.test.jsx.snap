// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Sword Art Online"
      src="https://sao.dev/"
    />
    <p>
      Sword Art Online
    </p>
  </div>
</div>
`;
